crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3340,11 +3637,62 @@ impl KnownFunction {
             | Self::Final
             | Self::NoTypeCheck
             | Self::RevealType
+            | Self::GetattrStatic
             | Self::StaticAssert => ParameterExpectations::AllValueExpressions,
         }
     }
 }
 
+/// This type represents bound method objects that are created when a method is called
+/// on an instance of a class. For example, the expression `Path("a.txt").touch` creates
+/// a bound method object that represents the `Path.touch` method which is bound to the
+/// instance `Path("a.txt")`.
+#[salsa::tracked]
+pub struct BoundMethodType<'db> {
+    /// The function that is being bound. Corresponds to the `__func__` attribute on a
+    /// bound method object
+    pub(crate) function: FunctionType<'db>,
+    /// The instance on which this method has been called. Corresponds to the `__self__`
+    /// attribute on a bound method object
+    self_instance: Type<'db>,
```

Comment:
Is this guaranteed to be an `InstanceType` or could it be something else? If not, should this be `InstanceTYpe` instead of `Type`?